core/indexer/elasticIndexer.go

```diff
@@ -0,0 +1,461 @@
+	txTemplate := getTemplateByName(txIndex, indexTemplates)
+	if txTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(txIndex, txTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	blocksTemplate := getTemplateByName(blockIndex, indexTemplates)
+	if blocksTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(blockIndex, blocksTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	miniblocksTemplate := getTemplateByName(miniblocksIndex, indexTemplates)
+	if miniblocksTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(miniblocksIndex, miniblocksTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	tpsTemplate := getTemplateByName(tpsIndex, indexTemplates)
+	if tpsTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(tpsIndex, tpsTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	ratingTemplate := getTemplateByName(ratingIndex, indexTemplates)
+	if ratingTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(ratingIndex, ratingTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	roundsTemplate := getTemplateByName(roundIndex, indexTemplates)
+	if ratingTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(roundIndex, roundsTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	validatorsTemplate := getTemplateByName(validatorsIndex, indexTemplates)
+	if ratingTemplate != nil {
+		err := ei.elasticClient.CheckAndCreateTemplate(validatorsIndex, validatorsTemplate)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ei *elasticIndexer) createIndexes() error {
+	firstTxIndexName := fmt.Sprintf("%s-000001", txIndex)
+	err := ei.elasticClient.CheckAndCreateIndex(firstTxIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstBlocksIndexName := fmt.Sprintf("%s-000001", blockIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstBlocksIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstMiniBlocksIndexName := fmt.Sprintf("%s-000001", miniblocksIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstMiniBlocksIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstTpsIndexName := fmt.Sprintf("%s-000001", tpsIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstTpsIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstRatingIndexName := fmt.Sprintf("%s-000001", ratingIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstRatingIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstRoundsIndexName := fmt.Sprintf("%s-000001", roundIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstRoundsIndexName)
+	if err != nil {
+		return err
+	}
+
+	firstValidatorsIndexName := fmt.Sprintf("%s-000001", validatorsIndex)
+	err = ei.elasticClient.CheckAndCreateIndex(firstValidatorsIndexName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ei *elasticIndexer) setInitialAliases() error {
+	firstTxIndexName := fmt.Sprintf("%s-000001", txIndex)
```

Comment:
could have iterate over a slice of strings. Could have had a variable called
```GO
var availableIndexes = []string{blockIndex, miniblocksIndex, txIndex...........}
```

and called them in a for range

Reply:
refactored